config/config.go

```diff
@@ -18,85 +19,69 @@ type TeamConfig struct {
 
 type Config struct {
 	Default_slack_channel string
+	Disable_slack         bool
+	Github_org            string
+	Slack_auth_token      string
+	Github_token          string
+	Quiet                 bool
+	Verbose               int
 	Severity              []SeverityConfig
 	Ecosystem             []EcosystemConfig
 	Team                  []TeamConfig
 }
 
-type Env struct {
-	GithubOrg      string `mapstructure:"GITHUB_ORG"`
-	SlackAuthToken string `mapstructure:"SLACK_AUTH_TOKEN"`
-	GithubToken    string `mapstructure:"GITHUB_TOKEN"`
-}
-
-var viperClient *viper.Viper
-
-type ViperParams struct {
-	ConfigPath  *string
-	Output      interface{}
-	EnvFileName *string
-}
-
-func getViper() *viper.Viper {
-	if viperClient == nil {
-		viperClient = viper.New()
+func fileExists(fname string) bool {
+	if _, err := os.Stat(fname); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
 	}
-	return viperClient
+	return true
 }
 
-func LoadConfig(params ViperParams) error {
+func GetUserConfig(configFile string) (Config, error) {
 	log := logger.Get()
 
-	v := getViper()
-
-	filename := filepath.Base(*params.ConfigPath)
-	extension := filepath.Ext(*params.ConfigPath)
-	configDir := filepath.Dir(*params.ConfigPath)
+	userCfg := Config{}
 
-	v.SetConfigName(filename)
-	v.AddConfigPath(configDir)
-	v.SetConfigType(strings.TrimLeft(extension, "."))
+	// Use reflection to register all config fields in Viper to set up defaults
+	cfgFields := reflect.ValueOf(userCfg)
+	cfgType := cfgFields.Type()
```

Comment:
this is really cool! how easily you can obtain information of a object and any other metadata you wanna get pull